ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx

```diff
@@ -15,45 +16,157 @@ import {
 import { MvlAttachment } from "../../../../types/mvlData";
 import { handleDownloadResult } from "../../../../utils";
 import { mvlPreferencesSetWarningForAttachments } from "../../../../store/actions";
+import customVariables from "../../../../../../theme/variables";
+import { H4 } from "../../../../../../components/core/typography/H4";
+import { showToast } from "../../../../../../utils/showToast";
 
-type Props = {
-  attachment: MvlAttachment;
+const BOTTOM_SHEET_HEIGHT = 375;
+
+const styles = StyleSheet.create({
+  activityIndicator: {
+    padding: 36
+  },
+  loaderContainer: {
+    display: "flex",
+    flex: 1,
+    alignItems: "center",
+    marginTop: BOTTOM_SHEET_HEIGHT / 7
+  }
+});
 
+type Props = {
   /**
    * Initial configuration for user preferences
    */
   initialPreferences: { dontAskAgain: boolean };
 
   /**
-   *  Called on right-button press with the user's selected preferences (see also {@link initialPreferences})
+   *  Called on right-button press with the user's selected preferences
+   *  (see also {@link initialPreferences}).
+   *  The promise is waited until fulfilled and a loader is shown.
    */
-  onConfirm: (preferences: { dontAskAgain: boolean }) => void;
+  onConfirm: (preferences: { dontAskAgain: boolean }) => Promise<void>;
 
+  /**
+   * The user canceled the action via the UI.
+   */
   onCancel: () => void;
+
+  /**
+   * When true, invoke onConfirm immediately and show a loader.
+   */
+  withoutConfirmation: boolean;
 };
 
-const DownloadAttachmentConfirmationBottomSheet = (
-  props: Props
-): React.ReactElement => {
+const DownloadAttachmentConfirmationBottomSheet = ({
+  initialPreferences,
+  onConfirm,
+  onCancel,
+  withoutConfirmation
+}: Props): React.ReactElement => {
   const [dontAskAgain, setDontAskAgain] = useState<boolean>(
-    props.initialPreferences.dontAskAgain
+    initialPreferences.dontAskAgain
+  );
+  const [isLoading, setIsLoading] = useState<boolean>(withoutConfirmation);
+  const [downloadError, setDownloadError] = useState<Error | null>(null);
+
+  const performDownload = useCallback(
+    () =>
+      onConfirm({ dontAskAgain }).catch((error: Error) => {
```

Comment:
Could it be a good idea move 
```typescript
setIsLoading(true);
setDownloadError(null);
```
inside the `performDownload` function?
I guess these actions should be always called on every download.
In this way we can remove the same functions calling in the other parts of the code

Reply:
I moved [setIsLoading](https://github.com/pagopa/io-app/pull/3706/commits/98709c6d6d3b566da81ae67ea8f87eef47d076bd#diff-04d9d7c779eec6038ab9af0b57e77a474cfeda11a4052b29b668461fcecb04f3R74) but not `setDownloadError` because the latter is specific to one scenario only.